File: plugins/absolute.py
import re
from plugin import Plugin


class AbsolutePlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'絶対値\s+(.+)', line)

        if not match:
            return None

        value = match.group(1)

        return f"print(abs({value}))"

File: web/plugins/absolute.js
import { WebPlugin } from "../engine/plugin.js";

export class AbsolutePlugin extends WebPlugin {
    constructor() {
        super("絶対値");
    }

    async execute(line, context) {
        const expression = line
            .replace(/^絶対値\s*/, "");

        const value = Number(
            context.parser.parseValue(
                expression,
                context.runtime
            )
        );

        context.runtime.write(
            Math.abs(value)
        );
    }
}